src/Raven.Server/Documents/ShardedHandlers/ShardedQueriesHandler.cs

```diff
@@ -0,0 +1,843 @@
+                                        val += v;
+                                }
+
+                                result[fld.Alias ?? fld.Name] = val;
+                                break;
+                            case AggregationOperation.None:
+                                break;
+                            default:
+                                throw new ArgumentException("Not supported " + fld.AggregationOperation);
+                        }
+                    }
+
+                    _result.Results.Add(_context.ReadObject(result, "reduced-result"));
+                }
+            }
+
+
+            public void ProjectAfterMapReduce()
+            {
+                if (_result.IsMapReduce == false
+                    || (_query.Metadata.Query.Select == null || _query.Metadata.Query.Select.Count == 0)
+                    && _query.Metadata.Query.SelectFunctionBody.FunctionText == null)
+                    return;
+
+                var fieldsToFetch = new FieldsToFetch(_query, null);
+                var retriever = new ShardedMapReduceResultRetriever(_parent.ShardedContext.ScriptRunnerCache, _query, null, fieldsToFetch, null, _context, false, null, null, null,
+                    _parent.ShardedContext.IdentitySeparator);
+
+                var currentResults = _result.Results;
+                _result.Results = new List<BlittableJsonReaderObject>();
+
+                foreach (var data in currentResults)
+                {
+                    (Document document, List<Document> documents) = retriever.GetProjectionFromDocument(new Document
+                    {
+                        Data = data
+                    }, null, null, fieldsToFetch, _context, null, CancellationToken.None);
+
+                    if (document != null)
+                    {
+                        AddDocument(document.Data);
+                    }
+                    else if (documents != null)
+                    {
+                        foreach (var doc in documents)
+                        {
+                            AddDocument(doc.Data);
+                        }
+                    }
+
+                    void AddDocument(BlittableJsonReaderObject documentData)
+                    {
+                        var result = _context.ReadObject(documentData, "modified-map-reduce-result");
+                        _result.Results.Add(result);
+                    }
+                }
+            }
+
+            private class GroupKey
+            {
+                private readonly List<object> _items;
+                private readonly int _hash;
+
+                public object this[int i] => _items[i];
+                public GroupKey(List<object> items)
+                {
+                    _items = items;
+                    _hash = items.Count;
+                    foreach (object item in items)
+                    {
+                        _hash = Hashing.Combine(_hash, item?.GetHashCode() ?? 0);
```

Comment:
Don't we need to use `ReduceKeyProcessor` to calculate the hash?

Reply:
Removed. Using now `AggregateOn` from `ReduceMapResultsOfAutoIndex`